### ISSUE TITLE
ChangeInventoryEvent.Held is fired wrongly

### ISSUE THREAD

**Issue by randombyte-developer** [SpongePowered/SpongeCommon#926]
Tested on SpongeVanilla `1.10.2-5.0.0-BETA-113`. `ChangeInventoryEvent.Held` is fired wrongly: The event is even thrown when the player just changes the selected slot although the slots are empty.
So scrolling through an empty hotbar with this code:

```
@Listener
void onChangeSelectedSlot(ChangeInventoryEvent.Held e, @Root Player e) {
    int i = player.getInventory().query<Hotbar>(Hotbar.class).getSelectedSlotIndex();
    player.sendMessage(Text.of(String.valueOf(i)));
}
```

results in printing the selected slot.
The event is named `...Held`, so I assume this should only be fired when the held item actually changes. When scrolling through an empty inventory, the held item never changes.

When the behavior gets corrected, please instantly add and implement an event for listening to selected hotbar slot changes.

**Comment by gabizou** [comment]
@randombyte-developer Can you provide a stack trace in cases for this to better help us debug and fix the problem?

**Comment by Zidane** [comment]
```
When the behavior gets corrected, please instantly add and implement an event for listening to selected hotbar slot changes.
```

We take PRs.

**Comment by randombyte-developer** [comment]
http://pastebin.com/RuRbyJUW The stacktrace

**Comment by ImMorpheus** [comment]
Working on this. PR scheduled for tomorrow.

EDIT:

Nevermind, I talked with Faith. Apparently the thing with ChangeInventoryEvent.Held is intended behaviour. He proposed a ChangeInventoryEvent.Held.Scroll, `Fired when a {@link Living} scroll through its {@link Hotbar}`. (specifically, every time the slot index change by 1 position). After implementing it, I discovered I could bypass the cancellation if I scrolled fast enough so there's something more to this. No PR, sorry :(.
